app/scripts/lib/stores/indexeddb-store.test.ts

```diff
@@ -12,111 +12,166 @@ describe('IndexedDBStore', () => {
+    beforeEach(() => {
+      // make indexedDb throw the FF DOMException `InvalidStateError`:
+      // "A mutation operation was attempted on a database that did not allow mutations."
+      indexedDB.open = (name: string, version?: number) => {
+        const request = {} as any;
+        if (name === dbName && version === dbVersion) {
+          // restore the original way so it works the second time.
+          indexedDB.open = originalOpen;
+
+          const error = new DOMException(
+            'A mutation operation was attempted on a database that did not allow mutations.',
+            'InvalidStateError',
+          );
+          request.error = error;
+          setTimeout(() => {
+            request.onerror?.({ target: request });
+          }, 0);
+        }
+        return request;
+      };
+    });
+
+    afterEach(() => {
+      indexedDB.open = originalOpen;
+    });
+
+    it('Falls back to an in-memory database', async () => {
+      db = new IndexedDBStore();
+      expect(db.isPersistedBackingStore).toBe(null);
       await db.open(dbName, dbVersion);
+      expect(db.isPersistedBackingStore).toBe(false);
       // Verify the database is open by performing an operation
       await db.set({ key: 'value' });
       const values = await db.get(['key']);
       expect(values).toEqual(['value']);
     });
+  });
 
-    it('rejects with TypeError for invalid version (0)', async () => {
-      await expect(db.open(dbName, 0)).rejects.toThrow(TypeError);
-    });
-
-    it('handles opening errors', async () => {
-      await db.open(dbName, dbVersion + 1);
-      db.close();
+  describe('indexedDB Available', () => {
+    beforeEach(() => {
       db = new IndexedDBStore();
-      // trigger a "VersionError" by trying to open the database with a
-      // an _earlier_ different version number
-      await expect(db.open(dbName, dbVersion)).rejects.toThrow(Error);
     });
 
-    it('handles database version upgrades', async () => {
-      await db.open(dbName, dbVersion);
-      await db.set({ key: 'value' });
-      db.close();
-      db = new IndexedDBStore();
-      await db.open(dbName, dbVersion + 1);
-      const [value] = (await db.get(['key'])) as string[];
-      expect(value).toBe('value');
+    afterEach(() => {
+      db?.close();
     });
 
-    it('does not reopen if already open', async () => {
-      const openSpy = jest.spyOn(indexedDB, 'open');
-      await db.open(dbName, dbVersion);
-      expect(openSpy).toHaveBeenCalled();
-      openSpy.mockRestore();
-      await db.open(dbName, dbVersion);
-      expect(openSpy).not.toHaveBeenCalled();
-      openSpy.mockRestore();
-    });
-  });
+    describe('open', () => {
+      it('opens the database successfully and creates store on first open', async () => {
+        expect(db.isPersistedBackingStore).toBe(null);
+        await db.open(dbName, dbVersion);
+        expect(db.isPersistedBackingStore).toBe(true);
```

Comment:
This new `describe` block and one test for `isPersistedBackingStore === true` are the only changes from here on down. The rest is just indentation.